pkg/controller/mongodb/mongodb_tls.go

```diff
@@ -17,17 +17,14 @@ import (
 	mdbv1 "github.com/mongodb/mongodb-kubernetes-operator/pkg/apis/mongodb/v1"
 )
 
-type tlsValidationResult struct {
-	valid   bool
-	message string
-}
-
 // validateTLSConfig will check that the configured ConfigMap and Secret exist and that they have the correct fields.
-// If TLS is correctly configured, the function will return a result where result.valid = true.
-// If it's incorrectly configured the result will instead have result.valid = false and a message for the user in result.message.
-func (r *ReplicaSetReconciler) validateTLSConfig(mdb mdbv1.MongoDB) (tlsValidationResult, error) {
+// The possible return values are:
+// - (true, nil) if the config is valid
+// - (false, nil) if the config is not valid
+// - (_, err) if an error occured when validating the config
+func (r *ReplicaSetReconciler) validateTLSConfig(mdb mdbv1.MongoDB) (bool, error) {
```

Comment:
the name and return value make it so clear now :)

Reply:
Agreed :)